Library/Homebrew/utils/curl.rb

```diff
@@ -14,7 +14,9 @@ module Curl
 
     module_function
 
-    def curl_executable
+    def curl_executable(use_brewed_curl: false)
+      return Formula["curl"].opt_bin/"curl" if use_brewed_curl
```

Comment:
Should this `raise` or similar if this file doesn't exist?

Reply:
I thought about it: I do raise `HomebrewCurlDownloadStrategyError` in resource_auditor and download_strategy. If I want to raise here, I can't use `HomebrewCurlDownloadStrategyError` as it needs to url, and I don't have the url in this method. So I can raise, but the error type will be different, if that is fine for you.